rudder-lang/src/compile.rs

```diff
@@ -8,29 +8,52 @@ use crate::{
     parser::{Token, PAST},
 };
 
-use typed_arena::Arena;
 use colored::Colorize;
 use std::{fs, path::Path};
+use typed_arena::Arena;
+use walkdir::WalkDir;
 
 /// Parses the whole stdlib
-pub fn parse_stdlib<'src>(past: &mut PAST<'src>, sources: &'src Arena<String>, libs_dir: &'src Path) -> Result<()> {
-    let core_lib = libs_dir.join("corelib.rl");
-    parse_file(past, sources, &core_lib, "corelib.rl")?;
-
-    let oses = libs_dir.join("oslib.rl");
-    parse_file(past, sources, &oses, "oslib.rl")?;
-
-    let cfengine_core = libs_dir.join("cfengine_core.rl");
-    parse_file(past, sources, &cfengine_core, "cfengine_core.rl")?;
+/// Parse all `.rl` files recursively to allow future layout changes.
+pub fn parse_stdlib<'src>(
+    past: &mut PAST<'src>,
+    sources: &'src Arena<String>,
+    libs_dir: &'src Path,
+) -> Result<()> {
+    fn is_rl_file(file: &Path) -> bool {
+        file.extension().map(|e| e == "rl").unwrap_or(false)
+    }
 
-    let stdlib = libs_dir.join("stdlib.rl");
-    parse_file(past, sources, &stdlib, "stdlib.rl")?;
+    let walker = WalkDir::new(libs_dir)
+        .into_iter()
+        .filter(|r| r.as_ref().map(|e| is_rl_file(e.path())).unwrap_or(true));
+    for entry in walker {
+        match entry {
+            Ok(e) => {
+                let path = e.path();
+                let filename = sources.alloc(
```

Comment:
this part could be in parse_file so that we can remove one parameter from it

Reply:
indeed, I wanted the keep the previous behavior of only passing the leaf file name for stdlib files, but if it does not matter it's indeed easier.

Reply:
the previous behavior is "historical" in the sense that is was mandatory before putting them in the Arena to have the proper lifetime for the file name

Reply:
:+1: